spec.bs

```diff
@@ -2942,17 +3006,29 @@ of the following global objects:
     1. If |result| is a [=ECMAScript/normal completion=]:
       1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
         |result|'s \[[Value]] to a {{GenerateBidOutput}}.
-      1. If no exception was [=exception/thrown=] in the previous step, then set |generatedBid| to
-        the result of [=converting GenerateBidOutput to generated bid=] with |generatedBidIDL|,
-        |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s
-        [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
-      1. Otherwise, set |generatedBid| to failure.
+      1. If no exception was [=exception/thrown=] in the previous step:
+        1. Let |bidOutput| to the result of [=converting GenerateBidOutput to generated bid=] with
+          |generatedBidIDL|, |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s
+          [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
+        1. If |bidOutput| is not failure, then set |generatedBid| to |bidOutput|.
+    1. Let |debugLossReportUrl| be |global|'s
+      [=InterestGroupBiddingAndScoringScriptRunnerGlobalScope/debug loss report url=] if it's not
+      failure, null otherwise.
+    1. Let |debugWinReportUrl| be null.
     1. If |generatedBid| is a [=generated bid=] and |generatedBid|'s [=generated bid/bid=]'s
-      [=bid with currency/value=] &le; 0, set |generatedBid| to failure.
-    1. If |generatedBid| is null, set it to failure.
-    1. If |generatedBid| is not failure:
+      [=bid with currency/value=] &le; 0, then set |generatedBid| to null.
+    1. If |generatedBid| is a [=generated bid=]:
       1. Set |generatedBid|'s [=generated bid/bid duration=] to |duration|.
       1. Set |generatedBid|'s [=generated bid/interest group=] to |ig|.
+      1. Set |debugWinReportUrl| to |global|'s
+        [=InterestGroupBiddingAndScoringScriptRunnerGlobalScope/debug win report url=] if it's not
+        failure, null otherwise.
+    1. Otherwise:
+      1. Set |generatedBid| to a new [=generated bid=] with the following [=struct/items=]:
+        : [=generated bid/no bid=]
+        :: true

```

Comment:
Why are we using an entirely different pattern for setting generatedBid fields here than from the non-otherwise case above?